Chessington.GameEngine/Pieces/Pawn.cs

```diff
@@ -15,27 +15,45 @@ public IEnumerable<Square> PawnMove(Board board, int direction, int startingRow)
             var currentRow = currentPos.Row;
             var currentCol = currentPos.Col;
 
-            if (!Square.At(currentRow + direction, currentCol).IsOccupied(board))
+            if (Square.At(currentRow + direction, currentCol).CanMoveTo(board))
             {
                 yield return Square.At(currentRow + direction, currentCol);
 
                 if (currentRow == startingRow &&
-                    !Square.At(currentRow + 2 * direction, currentCol).IsOccupied(board))
+                    Square.At(currentRow + 2 * direction, currentCol).CanMoveTo(board))
                 {
                     yield return Square.At(currentRow + 2 * direction, currentCol);
                 }
             }
         }
+
+        public IEnumerable<Square> PawnTake(Board board, int direction)
+        {
+            var currentPos = board.FindPiece(this);
+            var currentRow = currentPos.Row;
+            var currentCol = currentPos.Col;
+
+            var diagonallyRight = Square.At(currentRow + direction, currentCol + 1);
+            var diagonallyLeft = Square.At(currentRow + direction, currentCol - 1);
+            if (diagonallyRight.CanMoveOrTake(board, this) && diagonallyRight.IsOccupied(board))
+            {
+                yield return Square.At(currentRow + direction, currentCol + 1);
+            }
+            if (diagonallyLeft.CanMoveOrTake(board, this) && diagonallyLeft.IsOccupied(board))
```

Comment:
Yes, so it's more apparent here. Board consists of squares. So it makes more sense if Board handles check on squares, rather than passing whole board into square class